projects/packages/my-jetpack/src/js-static/jp-js-error-tracker/index.js

```diff
@@ -0,0 +1,323 @@
+/**
+ * JPJSErrorTracker - Simplified JavaScript error tracking library
+ * Captures all JavaScript errors and passes them to a callback
+ * @param {object} global - The global object (window in browsers)
+ */
+( function ( global ) {
+	/**
+	 * JPJSErrorTracker constructor
+	 * @param {Function} onError - Callback function that receives error data
+	 */
+	function JPJSErrorTracker( onError ) {
+		if ( typeof onError !== 'function' ) {
+			throw new Error( 'JPJSErrorTracker requires an onError callback function' );
+		}
+
+		this.onError = onError;
+		this.sessionId = this.generateSessionId();
+		this.pageLoadTime = Date.now();
+		this.errorCount = 0;
+		this.throttledErrors = new Map();
+		this.throttleMs = 1000; // Fixed throttle time
```

Comment:
The docs indicate this is configurable, but it seems to be hard-coded? Also, it might be wise to be a bit more conservative at 5s to start.

Reply:
Updated the code to reflect the docs